internal/controllers/app_controller.go

```diff
@@ -337,16 +407,34 @@ func (r *AppReconciler) reconcile(ctx context.Context, app *ketchv1.App, logger
 		// use latest deployment and watch events for each process
 		latestDeployment := app.Spec.Deployments[len(app.Spec.Deployments)-1]
 		for _, process := range latestDeployment.Processes {
-			var dep appsv1.Deployment
-			if err := r.Get(ctx, client.ObjectKey{
-				Namespace: framework.Spec.NamespaceName,
-				Name:      fmt.Sprintf("%s-%s-%d", app.GetName(), process.Name, latestDeployment.Version),
-			}, &dep); err != nil {
+
+			cli, err := kubernetes.NewForConfig(r.Config)
+			if err != nil {
+				return appReconcileResult{
+					err: err,
+				}
+			}
+
+			appType := "Deployment"
+			if app.Spec.Type != nil {
+				appType = *app.Spec.Type
+			}
+
+			wc := workloadClient{
+				k8sClient:         cli,
+				workloadName:      fmt.Sprintf("%s-%s-%d", app.GetName(), process.Name, latestDeployment.Version),
+				workloadNamespace: framework.Spec.NamespaceName,
+				workloadType:      appType,
```

Comment:
it'll be possible to use `workloadType: app.Spec.GetType()` there

Reply:
Changed